OpenRA.Mods.Common/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -130,13 +130,13 @@ public override bool HandleMouseInput(MouseInput mi)
 						var unit = World.ScreenMap.ActorsAtMouse(mousePos)
 							.WithHighestSelectionPriority(mousePos);
 
-						if (unit != null && unit.Owner == (World.RenderPlayer ?? World.LocalPlayer))
+						if (unit != null && unit.Owner.IsAlliedWith(World.RenderPlayer))
```

Comment:
Likewise here, the fallback is needed for players with the visibility cheat enabled.